platform/darwin/src/MGLStyleLayer.mm.ejs

```diff
@@ -21,23 +22,39 @@
 @end
 
 @implementation MGL<%- camelize(type) %>StyleLayer
+{
+    std::unique_ptr<mbgl::style::<%- camelize(type) %>Layer> _pendingLayer;
+}
 
 <% if (type == 'background') { -%>
 - (instancetype)initWithIdentifier:(NSString *)identifier
 {
     if (self = [super initWithIdentifier:identifier]) {
-        _layer = new mbgl::style::<%- camelize(type) %>Layer(identifier.UTF8String);
+        [self commonInit:identifier];
     }
     return self;
 }
+- (void)commonInit:(NSString *)identifier
+{
+    auto layer = std::make_unique<mbgl::style::<%- camelize(type) %>Layer>(identifier.UTF8String);
+    _pendingLayer = std::move(layer);
+    self.layer = _pendingLayer.get();
+}
 <% } else { -%>
 - (instancetype)initWithIdentifier:(NSString *)identifier source:(MGLSource *)source
 {
     if (self = [super initWithIdentifier:identifier source:source]) {
-        _layer = new mbgl::style::<%- camelize(type) %>Layer(identifier.UTF8String, source.identifier.UTF8String);
+        [self commonInit:identifier source:source];
```

Comment:
Not a big deal but the `commonInit` pattern is typically used when there is some shared logic that needs to be called by several different initializers and/or something that happens in init also happens latter on. In this case it might be clearer to just do the work in `initWithIdentifier:source:` itself.